jobs/furnishings/src/furnishings/stage_processors/post_processor.py

```diff
@@ -152,10 +87,119 @@ def get_info_by_name(name: Furnishing.FurnishingName) -> dict:
+            private_key=app.config.get('BCLAWS_SFTP_PRIVATE_KEY'),
+            private_key_algorithm=app.config.get('BCLAWS_SFTP_PRIVATE_KEY_ALGORITHM'),
+            private_key_passphrase=app.config.get('BCLAWS_SFTP_PRIVATE_KEY_PASSPHRASE')
+        )
+
+    @staticmethod
+    def _format_furnishings(furnishings_dict: dict, processed_date: datetime) -> dict:
+        """Format furnishing details presented in XML file."""
+        xml_data = {
+            'furnishings': {}
+        }
+        for name, furnishings in furnishings_dict.items():
+            xml_data['furnishings'][name] = XmlMeta.get_info_by_name(name)
+            xml_data['furnishings'][name]['items'] = sorted(furnishings, key=lambda f: f.business_name)
+        # we leave date and volume in XML blank
+        xml_data['effective_date'] = processed_date.strftime(XML_DATE_FORMAT)
+        return xml_data
+
+    @staticmethod
+    def _build_xml_data(xml_data: dict):
+        """Build XML payload."""
+        template = Path(
+            f'{current_app.config.get("TEMPLATE_PATH")}/gazette-notice.xml'
+        ).read_text()
+        jinja_template = Template(template, autoescape=True)
+
+        return jinja_template.render(xml_data)
+
+    @staticmethod
+    def _save_xml_payload(payload):
+        """Save XML payload."""
+        xml_payload = XmlPayload(payload=payload)
+        xml_payload.save()
+        furnishing_group = FurnishingGroup(xml_payload_id=xml_payload.id)
+        furnishing_group.save()
+        return furnishing_group, xml_payload
+
+    def update_furnishings_status(self, funishing_status, furnishing_group_id=None, notes=None):
+        """Update furnishing entries after processing."""
+        for furnishings in self._furnishings_dict.values():
+            for furnishing in furnishings:
+                if furnishing_group_id:
+                    furnishing.furnishing_group_id = furnishing_group_id
+                    furnishing.processed_date = datetime.utcnow()
+
+                if notes:
+                    furnishing.notes = notes
+
+                furnishing.status = funishing_status
+                furnishing.last_modified = datetime.utcnow()
+                furnishing.save()
+
+    def process(self):
+        """Postprocess to generate and upload file to external resources (BC Laws)."""
+        if not self._furnishings_dict:
+            return
+
+        # Create the XML data from the furnishings dict
+        xml_data = self._format_furnishings(self._furnishings_dict, self._processed_date.strftime('%I:%M %p'))
+        self._app.logger.debug('Formatted furnishing details presented in XML file')
+
+        # SFTP to BC Laws
+        payload = self._build_xml_data(xml_data)
+        filename = f'QP_CORP_{LegislationDatetime.format_as_legislation_date(self._processed_date)}.xml'
+
+        with self._bclaws_sftp_connection as client:
+            resp = client.putfo(
+                    fl=StringIO(payload),
```

Comment:
We can use `StringIO` as an in-memory file-like object so that we can upload to the sftp without having to save a file locally first.